mailpoet/tests/DataFactories/Subscriber.php

```diff
@@ -124,6 +124,7 @@ public function create(): SubscriberEntity {
 
     foreach ($this->segments as $segment) {
       $subscriberSegment = new SubscriberSegmentEntity($segment, $subscriber, 'subscribed');
+      $subscriber->getSubscriberSegments()->add($subscriberSegment);
```

Comment:
This makes the subscriber factory match `\MailPoet\Subscribers\SubscriberSegmentRepository::createOrUpdate`. Prior to this change, calling subscriber->getSegments() after using this factory would always return an empty collection, even after using the `withSegments` method.

I'm no longer using a check that is affected by this, but I thought it would be good to fix this inconsistency in general.